### PR TITLE
Prepare v1.0.0/0.24.0 releases

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Prepares the 1.0.0/0.24.0-beta releases.

## Short description of the changes
- Updates main package version to 1.0.0 and prerelease packages to v0.24.0-beta
- Adds changelog entry

### CODE REVIEW

**Comment by cartermp** [comment]
Hmm, let's hold off on this just yet. I think we should do a sweep of how we name things first, since we can't change those after 1.0.

**Comment by cartermp** [comment]
Issue here: https://github.com/honeycombio/honeycomb-opentelemetry-dotnet/issues/274

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+
+### Enhancements
+
+- Add support for `OTEL_SERVICE_NAME` environment variable (#245) | [@pkanal](https://github.com/pkanal)
+- Attempt to match Java distro resources (#249) | [@cartermp](https://github.com/cartermp)
+- Don't bundle instrumentation packages in distro (#250) | [@MikeGoldsmith](https://github.com/MikeGoldsmith)
+- Support debug option (#254) | [@cartermp](https://github.com/cartermp)
+- Add AspNetCore instrumentation extension to add baggage to new spans (#255) | [@MikeGoldsmith](https://github.com/MikeGoldsmith)
+- Set options with environment variables if they're present (#261) | [@cartermp](https://github.com/cartermp)
+- Add Honeycomb.OpenTelemetry.Autoinstrumentations #270 | [@cartermp](https://github.com/cartermp)
```
Comment (pkanal) [inline CHANGELOG.md:15]: We need to add #279 & #284 here I think

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+
+### Maintenance
+
+- Specify OTel version in README (#246) | [@pkanal](https://github.com/pkanal)
+- Add grpc smoke tests (#264) | [@pkanal](https://github.com/pkanal)
+- Add http/protobuf smoke tests (#263) | [@pkanal](https://github.com/pkanal)
```
Comment (pkanal) [inline CHANGELOG.md:21]: Smoke tests are available now for http/protobuf & grpc and also for metrics

**Review by pkanal — CHANGES_REQUESTED** [review]
I think we're ready to release! 🎉 I left a comment about a couple of PRs to add to the Enhancements but other than that I think we're good to go.

**Review by pkanal — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**